src/cudamatrix/cu-array.h

```diff
@@ -189,8 +189,8 @@ class CuSubArray: public CuArrayBase<T> {
   CuSubArray(const T* data, MatrixIndexT length) {
     // Yes, we're evading C's restrictions on const here, and yes, it can be used
     // to do wrong stuff; unfortunately the workaround would be very difficult.
-    CuArrayBase<T>::data_ = const_cast<T*>(data);
-    CuArrayBase<T>::dim_ = length;
+    this->data_ = const_cast<T*>(data);
```

Comment:
I don't think you should be making these kinds of changes in your PR. I understand that both of these constructs are the same, but most people won't, and will be confused by this kind of change in a seemingly irrelevant PR.

Reply:
This is needed for successfully compilation CUDA related code on windows. They are absolutely relevant with this PR. There are some weird issue with two-phase name lookup in Windows version of nvcc.

Reply:
Oh, wow.  Silly Windows!

Reply:
To be clear, it is NVIDIA’s problem. Actually, I find msvc is much more reliable nowadays.